signer/src/storage/mod.rs

```diff
@@ -43,6 +43,19 @@ pub trait DbRead {
         &self,
     ) -> impl Future<Output = Result<Option<model::BitcoinBlockHash>, Error>> + Send;
 
+    /// Get the latest canonical bitcoin blocks.
+    fn get_latest_canonical_bitcoin_blocks(
```

Comment:
Are you sure that you need these functions? I was under the impression that much of the #380 filtering could in a query.

Reply:
This is the query that will let us get the last 11 blocks and then we can get the average time from them, how would we not need this?

Reply:
What query / filtering are you talking about?

Reply:
Instead of fetching the the blocks to compute the MTP associated with a deposit request, we can can write a query that returns the MTP. No need to fetch the blocks. One such postgres function is `percentile_disc`.

> This is the query that will let us get the last 11 blocks and then we can get the average time from them, how would we not need this?

So if we want a query that filters requests where the current timestamp is greater than the `MTP + 512 * X`, where `X` is associated with the locktime, we could do that filtering in the same query using `percentile_disc`.

Reply:
I'd rather have the query have a larger logical unit than be simply responsible for a single thing.

Here's what the final function should look like in the coordinator:

```rust
    async fn get_pending_requests(
        &mut self,
        bitcoin_chain_tip: &model::BitcoinBlockHash,
        aggregate_key: &PublicKey,
        signer_public_keys: &BTreeSet<PublicKey>,
    ) -> Result<Option<utxo::SbtcRequests>, Error> {
        
        // ...
        
        let maybe_median_block_time = self.context
            .get_storage()
            .get_latest_canonical_bitcoin_blocks(6)
            .await?
            .last()
            .map(|block| block.time_mined);

        let pending_deposit_requests = self
            .context
            .get_storage()
            .get_pending_accepted_deposit_requests(bitcoin_chain_tip, context_window, threshold)
            .await?
            .iter()
            .filter(|&request| {
                match request.lock_time {
                    LockTime::BlockHeight(height) => {
                        // ... Assess based on block height
                    }
                    LockTime::Time(time) => {
                        // ... Use median
                    }
                }
            });
            
        // ...
            
    }
```

Reply:
I'm think this approach is going to be way more complex than the SQL based approach. If you don't compute the MTP in the query, you'll need to do so with in code for each deposit request. Doing it all in SQL might be too complex as well, so maybe somewhere in between would be best.